File: 36. Valid Sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashSet<String> st = new HashSet<>();
        for(int i = 0; i < 9; i++)
        {
            for(int j = 0; j < 9; j++)
            {
                if(board[i][j] != '.')
                {
                if (!st.add("Row " + i + board[i][j]))
                    return false;
                if(!st.add("Col " + j + board[i][j]))
                    return false;
                if(!st.add("Row" + i / 3 + " Col " + j / 3 + board[i][j]))
                    return false;
                }
            }
        }
        return true;
    }
}

File: 36. Valid Sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row_set = set()
        col_set = set()
        cube_set = set()
        for i in range(9):
            
            for j in range(9):
                if board[i][j] != '.':
                    t_row = "row " + str(i) + " " + board[i][j]
                    t_col = "col " + str(j) + " " + board[i][j]
                    t_square = "cube " + str(i // 3) + str(j // 3) + " " + board[i][j]
                    # print(t_row)
                    # print(t_col)
                    # print(t_square)
                    if t_row in row_set or t_col in col_set or t_square in cube_set:
                        return False
                    row_set.add(t_row)
                    col_set.add(t_col)
                    cube_set.add(t_square)
        return True
